### DISCUSSION TITLE
Retrain or resume

### DISCUSSION THREAD

**Discussion by alexKup88** [SHI-Labs/StyleNAT#4] [General]
First, thanks for sharing.

Could please add a yml file for training or resuming (better)?

I think evaluate should be as:

```python
python main.py type=evaluate
```

**Comment by stevenwalton** [comment]
Hey, thanks for taking interest. I'm sorry I have been a little slow and haven't gotten the training version of the code out yet. I promise that it will be soon. I'm actually working on getting image editing going first and then I'll get this out.

For now though, the `conf/runs` directory actually has all the information you need. There is a slight error there with the generator learning rate (it will be corrected in the next push: G_lr = 0.00004). But to help in the mean time here's the command I used to run the experiment 

`train.py --local_rank=0 --batch 8 --wandb --size 256 --path /data/datasets/ffhq_256 --checkpoint_path Dilated-7s-b8-TrainingFID-Flips/checkpoints --sample_path Dilated-7s-b8-TrainingFID-Flips --eval_gt_path /data/datasets/ffhq_256/images --G_channel_multiplier 2 --bcr --D_lr 0.0002 --D_sn --G_lr 0.00004 --lr_decay --lr_decay_start_steps 775000 --eval_freq 5000 --save_freq 50000 --block_type nat --use_flip --project_name StyleNAT --run_name FFHQ256-Dilated-7s-b8-TrainingFID-Flips --ckpt Dilated-7s-b8-TrainingFID-Flips/checkpoints/475000.pt --iter 1000000`

We're actually doing a bit of a rebase here from `argparse` to `hydra` so that we can cleanly change the kernel and dilation sizes. When training the code originally I was just editing the generator file manually and I didn't want burden others with this hacky method (it can be hard to track too). (If you can wait <1 week I'll have it in. This is a holiday week so I'm a little slow)

How things work now is that the `runs` directory is going to hold all the information to reproduce a specific run where the main conf will hold the more universal functions. I may break out the types (like inference, evaluation, editing, and training) but I'm unsure at this point. I am very open to feedback and want to make this as easy as possible for the users. 

For FFHQ-256 this file is what you want https://github.com/SHI-Labs/StyleNAT/blob/main/conf/runs/ffhq_256.yaml

I hope this helps? And I appreciate both your interest and patience. Let me know if there's anything else you need for now.

**Reply by alexKup88** [reply]
Thanks for coming back. I couldn't find train.py:

```bash
./models/generator.py
./models/basic_layers.py
./models/stylenat.py
./models/discriminator.py
./src/inference.py
./src/evaluate.py
./dataset/dataset.py
./main.py
./utils/CRDiffAug.py
./utils/inception.py
./utils/fid_score.py
./utils/distributed.py
./utils/improved_precision_recall.py
./op/fused_act.py
./op/__init__.py
./op/upfirdn2d.py
```

Best

**Reply by stevenwalton** [reply]
I've submitted PR #5 and the training code is included. You'll find configurations for both FFHQ-{256,1024}. The checkpoints are also available in the results table. Please let me know how it goes.
